src/panpkg/api.nim

```diff
@@ -153,21 +157,13 @@ proc loadImage*(path: string): Image =
     "stb_image must always return an image with 4 channels, " &
     "please report this on pan's GitHub"
 
-  {.push checks: off.}
-
   # convert from RGBA to ARGB because cairo <3
-  for y in 0..<height:
-    for x in 0..<width:
-      let
-        r = channels * (x + y * width)
-        g = r + 1
-        b = g + 1
-        a = b + 1
-      (img[r], img[g], img[b], img[a]) =
-        when cpuEndian == bigEndian: (img[a], img[r], img[g], img[b])
-        else: (img[b], img[g], img[r], img[a])
-
-  {.pop.}
+  # A somewhat crude check to dermine if the image is a png. If so
+  # then we need to pre-multiply alpha as PNGs are not pre-multipled
+  # but Cairo requires it.
+  let (_, _, ext) = os.splitFile(path)
+  let premultiply = ext.toLower == ".png"
```

Comment:
Nice of you to leave this note; I think the "better" way here would be to, instead of directly using `stbi.load` to load the image from disk implicitly:

1. Read the file from disk using [`readFile`](https://nim-lang.org/docs/io.html#readFile%2Cstring)
2. Check that the first few bytes match the PNG's magic byte sequence - `89 50 4e 47 0d 0a 1a 0a`

This magic sequence is listed in the sidebar on the [Wikipedia page for PNG](https://en.wikipedia.org/wiki/Portable_Network_Graphics).